### PR TITLE
 Add custom data field and fix json assemble

### PR DESCRIPTION
**Pull request by aleklisi**

### CODE REVIEW

File: lib/sparrow/apns.ex [hunk 1]
```diff
@@ -159,6 +151,19 @@ defmodule Sparrow.APNS do
     Sparrow.APNS.Errors.get_error_description(status, code)
   end
 
+  @spec make_body(Sparrow.APNS.Notification.t()) :: map
+  defp make_body(notification) do
+    alert = notification.alert_opts |> Map.new()
```
Comment (rslota) [inline lib/sparrow/apns.ex]:
```
alert = 
  notification.alert_opts 
  |> Map.new()
```

File: lib/sparrow/apns.ex [hunk 2]
```diff
@@ -159,6 +151,19 @@ defmodule Sparrow.APNS do
...
+
+    aps_opts =
+      notification.aps_dictionary_opts
+      |> Map.new()
+      |> Map.put("alert", alert)
+
+    [{"aps", aps_opts} | notification.custom_data]
```
Comment (rslota) [inline lib/sparrow/apns.ex]:
```
notification.custom_data
|> Map.new()
|> Map.put("aps", aps_opts)
```

Comment (rslota) [inline lib/sparrow/apns.ex]: Actually, why don't we just make `notification. aps_dictionary_opts` and `notification.custom_data` maps?

Comment (aleklisi) [reply lib/sparrow/apns.ex]: ??

Comment (aleklisi) [reply lib/sparrow/apns.ex]: ok I get it

**Pull request merged**